Feature/bavfw release engineer tooling.part2

rretanubun opened the pull request:
# Primary Changes
- Pull in the `brd21-swu-ota-generator` as a `git submodule`
- Add `prepare-signed-flasher-bundle.sh` and update the `README` with a one-shot sample of generating a signed flasher bundle tar from an unsigned flasher bundle tar.

# Dev Testing
- Tested by automating the `v0.9.2 unsigned flasher bundle` that I did manually during the v0.9.2 release process.

rretanubun committed:
BAVFW-1585: Add placeholder for release-engineer-tools

Git does not add empty sub-directory so we make a placholder
file for now.

rretanubun committed:
BAVFW-1585: Stage brd21-swu-ota-generator as git submodule

rretanubun committed:
BAVFW-1585: prepare-signed-flasher-bundle.sh added

This helper script should be a one-stop invocation
that lets the release-engineer take in a input flasher bundle
and then generate an updated version with

1. HSM Signed bootloader binary (with PIN prompt)
2. HSM Signed application binary (with PIN prompt)
3. Encrypted SWU OTA binary (using the public key from flasher-bundle)

mmb-davidsmith commented on release-engineer-tools/prepare-signed-flasher-bundle.sh:
```diff
@@ -0,0 +1,204 @@
+# ------------------------------------------------------------------------------
+# 1. Make sure you have the HSM Key plugged in
+# 2. Make sure you remember your HSM PIN
+# 3. To re-test the connection to the HSM, you can do:
```
Do you mean re-test or reset?

rretanubun replied:
`both` 😅 . `gpgconf --kill all && sudo systemctl restart pcscd` reset the daemon and `pkcs11-tool -O` re-test in case the HSM is not routed to the VM (for example)

mmb-davidsmith commented on release-engineer-tools/prepare-signed-flasher-bundle.sh:
```diff
@@ -0,0 +1,204 @@
+# ------------------------------------------------------------------------------
+
+if [ "$#" -ne 5 ]; then
+    printf "Arguments Count: $# is insufficient\n\n"
```
Given you're doing a `!=` check, it may make more sense to indicate that the script expects `exactly` 5 arguments, vs `at least` 5 arguments.

mmb-davidsmith replied:
Having said that, you've got the hard coded `verboseArg` below, which I'm guessing you intend to convert to an argument eventually. If that's the case, this should probably change somehow. In addition, I'm not sure how well positional arguments will work if you try to add a `-v` style option to configure verbosity as a user might do

```
tme.sifton prod.secure 0.9.4 bundle-unsigned.tar bundle-signed.tar -v
```

or

```
-v tme.sifton prod.secure 0.9.4 bundle-unsigned.tar bundle-signed.tar
```

rretanubun replied:
Fair point, I pushed a change that reworded this and ask for exactly the number of args.
I was going back on forth on weather this should be done using the `--key=value` style parser (which can be re-ordered) and decided against it since it makes it less things to type for the `release engineer`. I can be convinced otherwise, though.

rretanubun committed:
prepare-signed-flasher-bundle.sh: Updates and clarifications

Dry-run result with v0.9.4 release.

Since the next step in the release process is looking for
$signedFlasherBundle in specific filename format, lets
auto-generate the tar filename rather than having the user type it.

Also incorporated code-review feedback to state that
we need arguments in exactly the number and order needed.
